### PR TITLE
Why and when you should use kubernetes

### LINKED ISSUE #2440: Why and when you should use Kubernetes?
### Proposed title of article
Why and when you should use Kubernetes?

### Introduction:
Kubernetes is a powerful container management tool that automates the deployment and management of containers. Kubernetes is the next big wave in cloud computing and it’s easy to see why as businesses migrate their infrastructure and architecture to reflect a cloud-native, data-driven era.
Before the business decides to use Kubernetes, the question of why or when Kubernetes should be used arises. This article will address the issue.

### Key takeaways: 
At the end of this article the learner will be able to:
- understand the overview of Kubernetes and container orchestration
- know the top reasons why a business should go for Kubernetes and the benefits it offers
- understand the instances when Kubernetes can be used or not used

### References: 
N/A

### PR DESCRIPTION
**Pull request by erixbiz**
This fixes #2440

Wrote a short article on Why-and-when-you-should-use-Kubernetes. The Key takeaways being:-

1. understand the overview of Kubernetes and container orchestration.
2. know the top reasons why a business should go for Kubernetes and the benefits it offers.
3. understand the instances when Kubernetes can be used or not used.

Below is the link to the images that I have used in my article 
Why-and-when-you-should-use-Kubernetes/image-name/container.jpg

My author's folder is erick-wekesa

### CODE REVIEW

**Comment by BrianaNzivu** [comment]
Hey @mbaabuones 👋🏾. 
Please note that this is a duplicate of #2663.
You can proceed with the review regardless.
Have a good day.

**Comment by mbaabuones** [comment]
> 
> 
> Hey @mbaabuones 👋🏾.
> Please note that this is a duplicate of #2663.
> You can proceed with the review regardless.
> Have a good day.

@BrianaNzivu Thank you for bringing this to my notice. I was not aware. Thanks for allowing me to work on it. Good day.

File: content/articles/Why-and-when-you-should-use-Kubernetes/index.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+![WHY AND WHEN SHOULD WE USE KUBERNETES?](/Why-and-when-you-should-use-Kubernetes/image-name/container.jpg)
+# WHY AND WHEN SHOULD WE USE KUBERNETES?
```
Comment (mbaabuones) [inline content/articles/Why-and-when-you-should-use-Kubernetes/index.md:2]: Remove this header. Use the suggested structure.

File: content/articles/Why-and-when-you-should-use-Kubernetes/index.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+One can run many containers on a single virtual machine. This is like a virtual machine. It is possible to set and run many virtual machines on one single machine. This is a physical machine. Docker is an example of a container. It is one of the most popular containers used.  
+
+# Difference Between Virtual Machine and Container.
+
+1) For VM, You will manage the operating system. For Container, there is no management of the operating system. 
+2) For VM, it is an instance of an operating system. For Container, it is lightweight. Meaning containers are built to be started, stopped, and scaled out. This is without human intervention. 
+3) For VM, it is static i.e. as application demand increases, one creates and deploys the virtual machine. For Container, it is more dynamic. They allow one to respond to change in demand. With a container, one can easily and quickly restart it in case of any kind of failure. E.g. Hardware Failure
+
+
```
Comment (mbaabuones) [inline content/articles/Why-and-when-you-should-use-Kubernetes/index.md:20]: Avoid unnecessary spacing. One space after the end of a section is enough.

File: content/articles/Why-and-when-you-should-use-Kubernetes/index.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ 
+
+# WHY AND WHEN SHOULD WE USE KUBERNETES?
```
Comment (mbaabuones) [inline content/articles/Why-and-when-you-should-use-Kubernetes/index.md:23]: Dont capitalize every word. Please look at how the section headers have been written in the published articles.

File: content/articles/Why-and-when-you-should-use-Kubernetes/index.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+
+ 
+
+# WHY AND WHEN SHOULD WE USE KUBERNETES? 
+
+It is important to understand the why. Why should I use Kubernetes in software development? What kind of problem will it solve? With this understanding, development teams, are able to ship quality software. They will also be able to apply best industry practices and use relevant tools.  This will also speed up the process required to ship a product. 
+
+Containers are a great development tool. It offers a good way to put together different components of the application and run. When you want to ensure that you have a quick and easy way of moving to production, then containers are great. Once a container is in production, it needs to be looked into to cut downtime. An example would be, if one of the containers is down, another one needs to be started promptly. This can be achieved using Kubernetes. It will automatically do this process on our behalf.
```
Comment (mbaabuones) [inline content/articles/Why-and-when-you-should-use-Kubernetes/index.md:27]: In this section, use the aforementioned structure guidelines.

File: content/articles/Why-and-when-you-should-use-Kubernetes/index.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+3) Useful when a need to manage container storage and network arises. Kubernetes persistent volumes can be used to present data storage to one or more containers. This configuration allows containers to read and write application data and persist this data across many port instances.
+4) Kubernetes has network plugins that provide capabilities such as exposing pods to the internet, load balancing traffic across many replicas of the pods, and helps in network isolation. 
+5)  Kubernetes has an API that can extend. If one wants to extend functionalities outside the ones provided, this is achievable.
+
+
+# When you shouldn’t use it.
+
+1) When one is developing a monolithic application. The components of the application developed are dependent on each other. This is from the input/output operation to the processing and rendering of the data. With containers, your application will be divided into separate independent components. 
+2) When you don't have the time it requires to learn how to use Kubernetes because it has a steep learning curve. Kubernetes is a massive tool that requires sufficient time to learn.
```
Comment (mbaabuones) [inline content/articles/Why-and-when-you-should-use-Kubernetes/index.md:44]: Use the aforementioned structure guidelines.

File: content/articles/Why-and-when-you-should-use-Kubernetes/index.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
+![WHY AND WHEN SHOULD WE USE KUBERNETES?](/Why-and-when-you-should-use-Kubernetes/image-name/container.jpg)
```
Comment (mbaabuones) [inline content/articles/Why-and-when-you-should-use-Kubernetes/index.md:1]: You can place the image link in the section 'Overview of Kubernetes and container orchestration.'

File: content/articles/Why-and-when-you-should-use-Kubernetes/index.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+# WHY AND WHEN SHOULD WE USE KUBERNETES? 
+
+Kubernetes, also known as K8s. Is an open-source system for automating deployment, scaling, and management of containerized applications. Let's dive into the below concepts first. We then look into why and when we should use Kubernetes.
+
+# WHAT IS A CONTAINER
+
+Let us first begin by understanding what containers are. Why so?  Modern applications are being built using containers which are a virtualization environment. Kubernetes is software used to deploy and manage these containers.  Containers are environments that have been virtualized like virtual machines. 
+
+# Similarities Between a Virtual Machine and A container 
+
```
Comment (mbaabuones) [inline content/articles/Why-and-when-you-should-use-Kubernetes/index.md:11]: Delete these two sections as advised in the above guidelines.

**Review by mbaabuones — CHANGES_REQUESTED** [review]
@erixbiz Thank you for submitting your article. I am pleased to welcome you to the Section Engineering Program. This article provides meaningful information regarding Kubernetes and why we should use them. It can be improved by modifying it based on the suggested changes. Please take note of the following points and ping me when you finish making the requested changes:
- Follow [this link](https://www.section.io/engineering-education/) to read recently published articles. Pay attention to the introduction of these articles, their structure, conclusion and overall quality.
- You made 10 commits. Subsequently, ensure you have one or few commits when making a pull request. Ensure you have made all adjustments to your article before uploading the index file. 
- Please upload a hero image for this article. This is a requirement every time you make a pull request. All hero images should be 600 by 400 pixels.
-  All markdown files for our program should have at least level three headers (###). If a section has level three headers (###), the sub-sections under this section will have level four headers (####).
- In my view, the similarities and differences between VM and containers should not be part of this article. Please focus on sections that directly correlate with the topic. 
- Remove any space immediately after a section topic.
- I suggest you use the following structure. Please add the missing parts accordingly.
** introduction section- Add the two introduction paragraphs you wrote in the topic suggestion form. No need for a header here. The next sections should have a header.
** An overview of Kubernetes and container orchestration- Give an overview of Kubernetes and container orchestration.
** Why businesses should use Kubernetes- In this section, list the key points using sub-sections. Give explanation in these sub-sections. Many online materials can give you adequate information on this. Gather enough information to add value to this article.
** When you should use Kubernetes- In this section, list the key points in bullet form (bold) and make explanation. You may also decide to list them using sub-sections. 
** When you shouldn't use Kubernetes- In this section, list the key points in bullet form (bold) and make explanation. You may also decide to list them using sub-sections. 
** Conclusion- Give a brief summary of your article and the concluding remarks.

**Comment by mbaabuones** [comment]
@erixbiz It is now two weeks since this article was reviewed. Please work on the suggested changes so that we can move to the next stage.

**Comment by erixbiz** [comment]
> @erixbiz It is now two weeks since this article was reviewed. Please work on the suggested changes so that we can move to the next stage.

I really apologize for the delay. Just a quick one, should I create a new pull request?

**Comment by hectorkambow** [comment]
> > @erixbiz It is now two weeks since this article was reviewed. Please work on the suggested changes so that we can move to the next stage.
> 
> I really apologize for the delay. Just a quick one, should I create a new pull request or do a Push to this branch?

You should be able to make the changes on this PR via your forked PR - but if that gets to messy - the easiest way would be to create a new PR. 

What ever is easiest for you

**Comment by erixbiz** [comment]
> > > @erixbiz It is now two weeks since this article was reviewed. Please work on the suggested changes so that we can move to the next stage.
> > 
> > 
> > I really apologize for the delay. Just a quick one, should I create a new pull request or do a Push to this branch?
> 
> You should be able to make the changes on this PR via your forked PR - but if that gets to messy - the easiest way would be to create a new PR.
> 
> What ever is easiest for you

Thank you

**Comment by erixbiz** [comment]
@mbaabuones I have done the corrections as requested. Thanks

**Comment by erixbiz** [comment]
@mbaabuones I am done with the changes. Thanks

**Comment by ninjaginja** [comment]
plagiarism check completed - all good
[2120420-c3082bc8d4e242e81d2f-1626795793541.pdf](https://github.com/section-engineering-education/engineering-education/files/6849619/2120420-c3082bc8d4e242e81d2f-1626795793541.pdf)

**Review by mbaabuones — APPROVED** [review]

**Comment by mbaabuones** [comment]
> 
> 
> @mbaabuones I am done with the changes. Thanks

@erixbiz Thank you. Please note the changes I have made.

**Review by mbaabuones — APPROVED** [review]

**Review by louisefindlay23 — APPROVED** [review]
Final review complete, @erixbiz. Great article. Be sure to look at the minor changes I made to inform future articles. Folder names shouldn't have capital letters in them.

Thanks, @mbaabuones for peer reviewing.

**Comment by erixbiz** [comment]
@louisefindlay23 Well noted with thanks.

@mbaabuones  it was great working with you on this article as well. Thanks.

**Pull request merged**